tests/lib/rules/arrow-parens.js

```diff
@@ -29,6 +29,9 @@ var valid = [
     { code: "() => {}", options: ["as-needed"], ecmaFeatures: { arrowFunctions: true } },
     { code: "a => {}", options: ["as-needed"], ecmaFeatures: { arrowFunctions: true } },
     { code: "a => a", options: ["as-needed"], ecmaFeatures: { arrowFunctions: true } },
+    { code: "([a, b]) => {}", options: ["as-needed"], ecmaFeatures: { arrowFunctions: true, destructuring: true } },
+    { code: "({ a, b }) => {}", options: ["as-needed"], ecmaFeatures: { arrowFunctions: true, destructuring: true } },
+    { code: "(a = 10) => {}", options: ["as-needed"], ecmaFeatures: { arrowFunctions: true, destructuring: true, defaultParams: true } },
```

Comment:
destructuring, defaultParams, anything else that we might need to test for this case?